packages/db/tables.sql

```diff
@@ -0,0 +1,75 @@
+CREATE SCHEMA nftstorage;
+
+CREATE TYPE nftstorage.pin_status AS ENUM (
+  'Queued',
+  'Pinning',
+  'Pinned',
+  'Failed'
+);
+
+CREATE TYPE nftstorage.service AS ENUM (
+  'Pinata',
+  'IpfsCluster'
+);
+
+CREATE TYPE nftstorage.upload_type AS ENUM (
+  'Car',
+  'Blob',
+  'Multipart'
+);
+
+CREATE TABLE IF NOT EXISTS nftstorage.user (
+  id BIGSERIAL PRIMARY KEY,
```

Comment:
regarding the new `id`, i still think that we dont need it. 
Even if we change the auth again whatever identifier we use can go into the `issuer`, this id would never be used for selects. We always either get the issuer from tokens or the auth session and those will never be this new auto generated `id`.

Right now we have issuer and sub because we (i did :/) did a mistake and the tokens and kvs may have one of those, and we can't control that so we check both. In the migration im gonna normalize the kv data to always use `issuer` but we can still have tokens with `sub` inside so we need to keep checking for both.

If we start using a new id, we will probably end up checking for `issuer`, `sub` and `id`.

Reply:
> regarding the new `id`, i still think that we dont need it.
> Even if we change the auth again whatever identifier we use can go into the `issuer`, this id would never be used for selects. We always either get the issuer from tokens or the auth session and those will never be this new auto generated `id`.

Can it go into `issuer`? What do we do with existing users?

I just think we should identify users _ourselves_ and _store_ identifiers we get from auth providers along with the user...so that switching auth provider is a simple case of check the right field. This way there's no migrations or lazy upgrades and no confusions to be had about whether the field stores one type of ID or another. We can even support multiple providers.

The other problem I have is that `issuer` does not sound like an ID and I think it will be confusing to have an `issuer`/`user_issuer` in another table.

Reply:
ok, lets use the auto gen id and maybe change issuer to `magic_link_id` and sub to `github_id` sounds good ?